example/misc/infinite_set.cpp

```diff
@@ -49,6 +50,11 @@ constexpr auto doubleton(X x, Y y) {
 
 namespace boost { namespace hana {
     template <>
+    struct SetTheoretic<infinite_set_tag> {
+        static constexpr bool value = true;
+    };
```

Comment:
Your documentation specifies a minimal complete definition including all four functions. I think this is what makes the concept useful, the guarantee that those functions work.